code/modules/mob/dead/observer/observer.dm

```diff
@@ -782,6 +782,35 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	if(isobserver(user) && check_rights(R_SPAWN))
 		change_mob_type( /mob/living/carbon/human , null, null, TRUE) //always delmob, ghosts shouldn't be left lingering
 
+/mob/dead/observer/proc/ondemand_possess(mob/living/target)
+	var/possessable = target.ondemand_possessable
+	if(!isnum(possessable) || possessable <= 0)
+		return
+	var/original_key = target.key
+	var/control_message = control_mob_message(target)
+	var/confirmation = tgalert(src, control_message, "[target] can be possessed", "Yes", "No", Timeout=FALSE)
+	if(confirmation == "Yes" && !QDELETED(target))
+		if(target.key == original_key)
+			possessable = target.ondemand_possessable
+			if(isnum(possessable) && possessable > 0)
```

Comment:
just `if(target.ondemand_possessable)`

Reply:
it's a number, not boolean, what if it ends up being -1?

Reply:
that's a bug then

Reply:
I understand now, sorry